src/commands/login.rs

```diff
@@ -0,0 +1,50 @@
+use anyhow::{bail, Context, Result};
+use clap::Args;
+use warg_client::RegistryUrl;
+
+use crate::keyring::set_auth_token;
+
+use super::CommonOptions;
+
+/// Manage auth tokens for interacting with a registry.
+#[derive(Args)]
+pub struct LoginCommand {
+    /// The common command options.
+    #[clap(flatten)]
+    pub common: CommonOptions,
+
+    /// The subcommand to execute.
+    #[clap(flatten)]
+    keyring_entry: KeyringEntryArgs,
+}
+
+#[derive(Args)]
+struct KeyringEntryArgs {
+    /// The URL of the registry to store an auth token for.
+    #[clap(value_name = "URL")]
+    pub url: Option<RegistryUrl>,
+}
+
+impl KeyringEntryArgs {
+    fn set_entry(&self, home_url: Option<String>, token: &str) -> Result<()> {
+        if let Some(url) = &self.url {
+            set_auth_token(url, token)
+        } else if let Some(url) = &home_url {
+            set_auth_token(&RegistryUrl::new(url)?, token)
+        } else {
+            bail!("Please configure your home registry: warg config --registry <registry-url>")
+        }
+    }
+}
+
+impl LoginCommand {
+    /// Executes the command.
+    pub async fn exec(self) -> Result<()> {
+        let token = rpassword::prompt_password("Enter auth token:\n")
```

Comment:
Not sure why we didn't use https://docs.rs/dialoguer/0.11.0/dialoguer/struct.Password.html before... :thinking:

I guess for consistency we should use either rpassword or dialoguer in both places, so up to you if you want to change both in this PR.